packages/open-payments/src/client/token.test.ts

```diff
@@ -0,0 +1,80 @@
+import { createTokenRoutes, rotateToken } from './token'
+import { OpenAPI, HttpMethod, createOpenAPI } from 'openapi'
+import config from '../config'
+import {
+  defaultAxiosInstance,
+  mockAccessToken,
+  mockOpenApiResponseValidators,
+  silentLogger
+} from '../test/helpers'
+import nock from 'nock'
+
+describe('token', (): void => {
+  let openApi: OpenAPI
+
+  beforeAll(async () => {
+    openApi = await createOpenAPI(config.OPEN_PAYMENTS_AS_OPEN_API_URL)
+  })
+
+  const axiosInstance = defaultAxiosInstance
+  const logger = silentLogger
+  const openApiValidators = mockOpenApiResponseValidators()
+
+  describe('createTokenRoutes', (): void => {
+    test('creates response validator for token requests', async (): Promise<void> => {
+      jest.spyOn(openApi, 'createResponseValidator')
+      createTokenRoutes({ axiosInstance, openApi, logger })
+
+      expect(openApi.createResponseValidator).toHaveBeenCalledTimes(1)
+      expect(openApi.createResponseValidator).toHaveBeenCalledWith({
+        path: '/token/{id}',
+        method: HttpMethod.POST
+      })
+    })
+  })
+
+  describe('rotateToken', (): void => {
+    test('returns accessToken if passes validation', async (): Promise<void> => {
+      const accessToken = mockAccessToken()
+
+      const manageUrl = new URL(accessToken.access_token.manage)
+      nock(manageUrl.origin).post(manageUrl.pathname).reply(200, accessToken)
+
+      const result = await rotateToken(
+        {
+          axiosInstance,
+          openApi,
+          logger
+        },
+        {
+          url: accessToken.access_token.manage,
+          accessToken: accessToken.access_token.value
```

Comment:
It doesn't really matter since we are throwing anyway, but I would put a different "fake" accessToken here because "accessToken.access_token.value" is the "new", rotated value

Reply:
Yeah, I got it.